server/src/main/kotlin/suwayomi/tachidesk/manga/impl/Chapter.kt

```diff
@@ -110,61 +111,83 @@ object Chapter {
     suspend fun fetchChapterList(mangaId: Int): List<SChapter> {
         val manga = getManga(mangaId)
         val source = getCatalogueSourceOrStub(manga.sourceId.toLong())
+        val isHttpSource = source is HttpSource
 
         val sManga =
             SManga.create().apply {
                 title = manga.title
                 url = manga.url
             }
 
-        val numberOfCurrentChapters = getCountOfMangaChapters(mangaId)
         val chapterList = source.getChapterList(sManga)
+        val now = Instant.now().epochSecond
+        val chaptersInDb =
+            transaction {
+                ChapterTable.select { ChapterTable.manga eq mangaId }
+                    .map { ChapterTable.toDataClass(it) }
+                    .toSet()
+            }
 
-        // Recognize number for new chapters.
-        chapterList.forEach { chapter ->
-            (source as? HttpSource)?.prepareNewChapter(chapter, sManga)
-            val chapterNumber = ChapterRecognition.parseChapterNumber(manga.title, chapter.name, chapter.chapter_number.toDouble())
-            chapter.chapter_number = chapterNumber.toFloat()
-        }
+        val chaptersToInsert = mutableListOf<ChapterDataClass>()
+        val chaptersToUpdate = mutableListOf<ChapterDataClass>()
+
+        chapterList.reversed().forEachIndexed { index, fetchedChapter ->
+            val chapterEntry = chaptersInDb.find { it.url == fetchedChapter.url }
+
+            val chapterData =
```

Comment:
```suggestion
val realUrl = if (isHttpSource) {
  (source as? HttpSource)?.getChapterUrl(fetchedChapter)
} else {
  null
}

val chapterData = ChapterDataClass.fromSChapter(
  fetchedChapter,
  chapterEntry?.id ?: 0,
  index + 1,
  now,
  mangaId,
  realUrl,
)
```

then the first `fromSChapter` is unused and can be removed